packages/babel-preset-env/src/polyfills/corejs3/usage-plugin.js

```diff
@@ -84,33 +86,56 @@ export default function(
       });
     },
 
+    // import('something').then(...)
     CallExpression({ node }) {
       if (t.isImport(node.callee)) {
         this.addUnsupported(PromiseDependencies);
       }
     },
 
-    // Symbol()
-    // new Promise
-    ReferencedIdentifier({ node: { name }, parent, scope }) {
-      if (t.isMemberExpression(parent)) return;
-      if (!has(BuiltIns, name)) return;
-      if (scope.getBindingIdentifier(name)) return;
-
-      const BuiltInDependencies = BuiltIns[name];
-      this.addUnsupported(BuiltInDependencies);
-    },
-
     // for-of loop
     ForOfStatement() {
       this.addUnsupported(CommonIterators);
     },
 
-    // spread
-    ArrayExpression({ node }) {
-      if (node.elements.some(el => t.isSpreadElement(el))) {
-        this.addUnsupported(CommonIterators);
+    ArrayPattern() {
+      // const [a, b] = c
+      this.addUnsupported(CommonIterators);
+    },
+
+    ObjectPattern({ node, parent, scope }) {
+      let canBeStatic, builtIn, right;
+      if (t.isVariableDeclarator(parent)) {
+        right = parent.init;
+      } else if (t.isAssignmentExpression(parent)) {
+        right = parent.right;
       }
+      if (right) {
+        builtIn = right.name;
+        canBeStatic =
+          has(StaticProperties, builtIn) &&
+          !scope.getBindingIdentifier(builtIn);
+      }
+
+      for (const { key } of node.properties) {
+        const name = t.isIdentifier(key)
+          ? key.name
+          : t.isStringLiteral(key)
+          ? key.value
+          : null;
```

Comment:
I think that this might have problems with things like
```js
let trim = "toString";
const { [trim]: toString } = ""; 
```
Since the key is an `Identifier`, it will add the polyfill for `"".trim`.

Maybe something like this?
```js
for (const { key, computed } of node.properties) {
  const name = !computed && t.isIdentifier(key)
      	         ? key.name
                 : t.isStringLiteral(key)
                 ? key.value
                 : null;
}
```

Reply:
I rewrote it in another way yesterday and will push it in an hour. Ralated the above comment too.